.github/workflows/go.yml

```diff
@@ -16,15 +16,22 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v3.5.0
       with:
-        go-version: 1.19.5
+        go-version: 1.22.1
+    
+    - name: Generate
+      run: go generate ./...
+        
+
+    - name: Lint
+      uses: golangci/golangci-lint-action@v4
+      with:
+        version: v1.56
 
     - name: Build
       run: go build -v ./cmd/server
 
     - name: Test
-      run: |
-        go generate ./...
-        go test -race -coverprofile=coverage.txt -covermode=atomic ./...
+      run: go test -race -coverprofile=coverage.txt -covermode=atomic ./...
```

Comment:
Is there a specific reason in generating code before running the linters? If I understand correctly from golangci-lint documentation, generated files are detected and errors won't be reported. And as those files are only required for testing, I think it's fine to do code generation down there :) 

[Here](https://golangci-lint.run/usage/configuration/) is the relevant golangci docs section:
```
  # Which files to exclude: they will be analyzed, but issues from them won't be reported.
  # There is no need to include all autogenerated files,
  # we confidently recognize autogenerated files.
  # If it's not, please let us know.
  # "/" will be replaced by current OS file path separator to properly work on Windows.
  # Default: []
  exclude-files:
    - ".*\\.my\\.go$"
    - lib/bad.go
 ```

Reply:
Ok, thanks for clarification :+1: